code401challenges/src/main/java/code401challenges/BinarySearch.java

```diff
@@ -0,0 +1,25 @@
+package code401challenges;
+
+public class BinarySearch {
+    //this is very helpful; https://en.wikipedia.org/wiki/Binary_search_algorithm
+    public static int binarySearch(int[] arrIn, int key){
+        int lower = 0;
+        int upper = arrIn.length - 1;
+        while(lower <= upper){
+            int mid = (lower + upper)/2;
+
+//            String out = String.format("Lower is %d Upper is %d Mid is %d", lower, upper, mid);
```

Comment:
I like that you are practicing String.format, I would remove this from the solution code though. We will be taking off points soon for "corpse code"